pkg/cloud/services/networking/router.go

```diff
@@ -229,26 +257,8 @@ func (s *Service) getRouterInterfaces(routerID string) ([]ports.Port, error) {
 	})
 }
 
-func (s *Service) getRouter(clusterName string) (routers.Router, subnets.Subnet, error) {
-	routerName := getRouterName(clusterName)
-	router, err := s.getRouterByName(routerName)
-	if err != nil {
-		return routers.Router{}, subnets.Subnet{}, err
-	}
-
-	subnetName := getSubnetName(clusterName)
-	subnet, err := s.getSubnetByName(subnetName)
-	if err != nil {
-		return router, subnets.Subnet{}, err
-	}
-
-	return router, subnet, nil
-}
-
-func (s *Service) getRouterByName(routerName string) (routers.Router, error) {
-	routerList, err := s.client.ListRouter(routers.ListOpts{
-		Name: routerName,
-	})
+func (s *Service) getRouterByFilter(opts routers.ListOpts) (routers.Router, error) {
```

Comment:
nit: This function has only 1 caller and it's not super complicated. Personally I'd either re-use it in the create flow, which would have the benefit of de-deduplicating the error message if there's more than one matching router, or inline it in the delete flow.

This is just a nit. Not important to change if you don't want.